packages/xo-web/src/common/xo/install-pool-patches-modal/index.js

```diff
@@ -0,0 +1,124 @@
+import _ from 'intl'
+import ActionButton from 'action-button'
+import Component from 'base-component'
+import Icon from 'icon'
+import React from 'react'
+import SingleLineRow from 'single-line-row'
+import Tooltip from 'tooltip'
+import { Container, Col } from 'grid'
+import { connectStore } from 'utils'
+import {
+  createCollectionWrapper,
+  createCounter,
+  createGetObject,
+  createGetObjectsOfType,
+  createSelector,
+} from 'selectors'
+import { forEach } from 'lodash'
+import { SelectSr } from 'select-objects'
+import { VmItem } from 'render-xo-item'
+import { ejectCd, isSrWritable, setDefaultSr } from 'xo'
+
+@connectStore(() => {
+  const getPool = createGetObject((_, props) => props.pool)
+  return {
+    pool: getPool,
+    poolMaster: createGetObject(
+      createSelector(getPool, ({ master }) => master)
+    ),
+    vbds: createGetObjectsOfType('VBD').filter(
+      createSelector(getPool, ({ id }) => vbd => vbd.$pool === id)
+    ),
+  }
+})
+export default class InstallPoolPatchesModalBody extends Component {
+  _getVmsWithCds = createSelector(
+    () => this.props.vbds,
+    createCollectionWrapper(vbds => {
+      const vmIds = []
+      forEach(vbds, vbd => {
+        if (
+          vbd.is_cd_drive &&
+          vbd.VDI !== undefined &&
+          !vmIds.includes(vbd.VM)
+        ) {
+          vmIds.push(vbd.VM)
+        }
+      })
+      return vmIds
+    })
+  )
+
+  _getNVmsWithCds = createCounter(this._getVmsWithCds)
+
+  _ejectCds = () => Promise.all(this._getVmsWithCds().map(ejectCd))
+
+  _getTooltip = createSelector(this._getVmsWithCds, vmIds =>
+    vmIds.map(vmId => (
+      <p className='m-0' key={vmId}>
+        <VmItem id={vmId} />
+      </p>
+    ))
+  )
+
+  render () {
+    const { pool, poolMaster } = this.props
+    const needDefaultSr =
+      poolMaster.productBrand !== 'XCP-ng' && pool.default_SR === undefined
+    const someCdsInserted = this._getNVmsWithCds() > 0
+
+    return (
+      <Container>
+        {!needDefaultSr &&
+          !someCdsInserted && (
+            <SingleLineRow>
+              <Col>{_('confirmPoolPatch')}</Col>
+            </SingleLineRow>
+          )}
+        {needDefaultSr && [
```

Comment:
This is better than ternary condition ?

Reply:
What would it look like? I'd need a double ternary operator here since there are 2 conditions.

Reply:
My bad you're right :+1: